src/document-ready.js

```diff
@@ -28,18 +28,18 @@ export function isDocumentReady(doc) {
 /**
  * Calls the callback when document is ready.
  * @param {!Document} doc
- * @param {!Function} callback
+ * @param {function(!Document)} callback
  */
 export function onDocumentReady(doc, callback) {
   let ready = isDocumentReady(doc);
   if (ready) {
-    callback();
+    callback(doc);
```

Comment:
[M̸̲͖̼̖̭̻̊̔̄͂͛̀̾́̐͡á̴̱̬̞̜̮͍̟̩̜̃̍̅̓͛y̶̢̘̩̱̮̥͙̳̣̑͊̅͊͗̍̎͝ H̭̩͎͙̬̭̻͈̻̅͋̆́̆̐̀͋̔̾e̺̗̤͓̜̝̭̓̎̐̀͊͂̊̔͌͟͝ Ẅ̢̯̬̘͓̩͙͕̼́̉͑̅̚͟h̷͖͉̳̠͔̠͌̋̅̾̉̑̅̚͠o̵̡̧̡͙͎̜̱̐̍̐̿̿́͒͐̂ W̵͚̳͔͇̝͍̙͎̤͈̓̎̈́̃͆ạ̵̢̯̩̑̑̍͂͜͝͡ͅi̷̢̢̨̗̞̼͖̠͂̔́̉̑͒̇̈͋͠ẗ̴̨̛̞̰̫̺̮̥͉̙̼̂͂̔̂͊̊s̷̛̫̦̠̝̥̬̀̓̔͛́͗͑͘͜͢͢ B͚͎̣̟̰͈̰͌́͐͘͘ę̖̦̩̱͔͎̤̊̊̑͊͌͒̾͘͟͠͞h̻͍͍̲̦̑͒̀̉̍ì̡̬̦̬̪͚̖̿̌̾̒͋̄͊̚͞n̡̧̛̮̝̰̣̥͓͓̅͗̀̆̾̒̎͗͟d̷̡͙̙̪̬̾͒͐͂̆ͅ t̡̨͉͔̻͐́͂̋̔̀͌͘̕͠h̵̲̟͉̞̝̮̞̝͓̼̅͂͊͆̽̄̔̇̕è̸̙̪̻̪̮̭̌͢͟͡͠͠ W̩̳̩͕̪̠̻̺̣̱͐̆͗̈́̄͂͒͆́ą̖̯͕̰͖̹̗͔͒͌͊͋̉̚͡ļ̩̺̗̠̯̐̓͗͑͘l̞̣͚͕̬̳̓̓̉̀͛̚͢ N̨̲̟͇̘̅̽́͊̀̓ē̘̺͉̗̪̝̪̫̋͗͆̌͛͑v̷̢̡̮̫͎̱͕̱̹̭̆͗̏̅́͗́̈́́͠e̵̹̮̗͈̹͂͂͆͗̀͊̊̕͠r̶̢̨͇̞̳̱͆̃̽̂̅̎́̑͡ L̗̩͕̠̼̅̆͗͆̃͞e̶̤̻̮̟̺̤̖̤̣̬͐̅͊̋͘a̴̢̺̫̯̻̫̺̲̔̌̾̃̾̐v̶̞̙̭̞̀͑̑̀̅͊͜͝ę̖͔̤̤͎͍͕̪̦͂̊̋͋̊̅͘ Ǫ̷͓̼̞͓̉͛̍̎͡ͅũ̶͉͙͖̞͉̺̻͒̏̑̊̍͘͘͞ŗ̶̡̠͓̩͚̱̒͛͌̋̌̎͐͜͝͝ͅ C̶̨͙̟͓̠̝̟̾̏̆̉̎̋̃̃͟͡͠ơ̴̤͕͉̟͉̬̲̪̦̪̈̂͆́͂́̆̿d̝̱̭͇̻̿̆̈̎́͌͆̊̊é̶̻͉͖̻̭̼̥̀̍̑̿̀͞b̴̧̤͎͎͎͇̗̈́̊̉̄͜͝ȧ̞̜͖̙͇̀͌̏̾͢s̵̨̪̞̭͎̲̜̥͍͍̀̄̏̒̐̐͗͘̕ḛ̸͎͖̠̼̲̫̺̰̅̽͊́̊̋̉̂͊͒͜
](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony)